tooling/test/sugar.go

```diff
@@ -208,6 +216,67 @@ func (e ExpectBuilder) BodyWithHint(hint string, body interface{}) ExpectBuilder
 	return e
 }
 
+func (e ExpectBuilder) Validate(t *testing.T, res *http.Response, localReport Reporter) {
+	t.Helper()
+
+	checks := validateResponse(t, e, res)
+	for _, c := range checks {
+		t.Run(c.testName, func(t *testing.T) {
+			if !c.checkOutput.Success {
+				localReport(t, c.checkOutput.Reason)
+			}
+		})
+	}
+}
+
+type AllOfExpectBuilder struct {
+	Expect_ []ExpectValidator `json:"expect,omitempty"`
+}
+
+var _ ExpectValidator = (*AllOfExpectBuilder)(nil)
+
+func AllOf(expect ...ExpectValidator) AllOfExpectBuilder {
+	return AllOfExpectBuilder{Expect_: expect}
+}
+
+func (e AllOfExpectBuilder) Validate(t *testing.T, res *http.Response, localReport Reporter) {
+	t.Helper()
+
+	for _, expect := range e.Expect_ {
+		expect.Validate(t, res, localReport)
+	}
+}
+
+type AnyOfExpectBuilder struct {
+	Expect_ []ExpectBuilder `json:"expect,omitempty"`
```

Comment:
Could we use a `ExpectValidator` here?

Reply:
It's doable, but not without some changes because the `AnyOfExpectBuilder`'s `Validate` function uses `validateResponse` which operates on an `ExpectBuilder` and returns the checks so they can be expected.

Up for proposals on changing the `Validate` function (or bundling everything into the `ExpectBuilder` struct) to support `AnyOf` working on multiple layers (i.e. on top of `AnyOf` or `AllOf`), although might also be ok to wait until we need it. WDYT?

Reply:
Sure, let's hold until we need it 👍